hazelcast/src/main/java/com/hazelcast/query/impl/predicates/EvaluatePredicate.java

```diff
@@ -72,13 +72,13 @@ public boolean apply(Map.Entry mapEntry) {
     }
 
     @Override
-    public Set<QueryableEntry> filter(QueryContext queryContext) {
-        Index index = queryContext.matchIndex(indexName, QueryContext.IndexMatchHint.EXACT_NAME);
+    public Set<QueryableEntry> filter(QueryContext queryContext, int ownedPartitionCount) {
+        Index index = queryContext.matchIndex(indexName, QueryContext.IndexMatchHint.EXACT_NAME, ownedPartitionCount);
```

Comment:
The implementation of this class looks suspicious to me. Evaluate predicate is an indexed predicate over some other predicate. `matchIndex` may return `null` if the target index is not ready, causing NPE during query execution. We need to add tests for bitmap indexes in addition to existing tests, to make sure that the logic works properly.